api/core/tools/provider/builtin/qrcode/tools/qrcode_generator.py

```diff
@@ -2,34 +2,69 @@
 import logging
 from typing import Any, Union
 
-import qrcode
+from qrcode.constants import ERROR_CORRECT_H, ERROR_CORRECT_L, ERROR_CORRECT_M, ERROR_CORRECT_Q
+from qrcode.image.base import BaseImage
 from qrcode.image.pure import PyPNGImage
+from qrcode.main import QRCode
 
 from core.tools.entities.tool_entities import ToolInvokeMessage
 from core.tools.tool.builtin_tool import BuiltinTool
 
 
 class QRCodeGeneratorTool(BuiltinTool):
+    error_correction_levels = {
+        'L': ERROR_CORRECT_L,  # <=7%
+        'M': ERROR_CORRECT_M,  # <=15%
+        'Q': ERROR_CORRECT_Q,  # <=25%
+        'H': ERROR_CORRECT_H,  # <=30%
+    }
+
     def _invoke(self,
                 user_id: str,
                 tool_parameters: dict[str, Any],
                 ) -> Union[ToolInvokeMessage, list[ToolInvokeMessage]]:
         """
             invoke tools
         """
-        # get expression
+        # get text content
         content = tool_parameters.get('content', '')
         if not content:
             return self.create_text_message('Invalid parameter content')
 
+        # get border size
+        border = tool_parameters.get('border', '')
+        if not border or isinstance(border, str):
+            return self.create_text_message('Invalid parameter border')
```

Comment:
There will be an error if border is 0, maybe we should check if it's a empty string

Reply:
Fixed. Thanks for the carefulness.
Considering the border is a number in runtime, I would skip the type checks and make user it's in the range of 0 to 100.